macaulay2 build failure - missing boost components

```
    [macaulay2-1.24.11]   [spkg-install] Cloning into 'M2/Macaulay2/editors/emacs'...
    [macaulay2-1.24.11]   [spkg-install] Cloning into 'M2/submodules/memtailor'...
    [macaulay2-1.24.11]   [spkg-install] Cloning into 'M2/submodules/mathic'...
    [macaulay2-1.24.11]   [spkg-install] Cloning into 'M2/submodules/mathicgb'...
    [macaulay2-1.24.11]   [spkg-install] Configuring macaulay2-1.24.11 with cmake
    [macaulay2-1.24.11]   [spkg-install] CMake Warning:
    [macaulay2-1.24.11]   [spkg-install]   Ignoring extra path from command line:
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install]    "/Users/runner/sage-local/var/tmp/sage/build/macaulay2-1.24.11/src/M2/BUILD/build"
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] CMake Warning:
    [macaulay2-1.24.11]   [spkg-install]   Ignoring extra path from command line:
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install]    "."
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] -- The C compiler identification is AppleClang 15.0.0.15000309
    [macaulay2-1.24.11]   [spkg-install] -- The CXX compiler identification is AppleClang 15.0.0.15000309
    [macaulay2-1.24.11]   [spkg-install] -- Detecting C compiler ABI info
    [macaulay2-1.24.11]   [spkg-install] -- Detecting C compiler ABI info - done
    [macaulay2-1.24.11]   [spkg-install] -- Check for working C compiler: /Users/runner/sage-local/libexec/ccache/gcc - skipped
    [macaulay2-1.24.11]   [spkg-install] -- Detecting C compile features
    [macaulay2-1.24.11]   [spkg-install] -- Detecting C compile features - done
    [macaulay2-1.24.11]   [spkg-install] -- Detecting CXX compiler ABI info
    [macaulay2-1.24.11]   [spkg-install] -- Detecting CXX compiler ABI info - done
    [macaulay2-1.24.11]   [spkg-install] -- Check for working CXX compiler: /Users/runner/sage-local/libexec/ccache/g++ - skipped
    [macaulay2-1.24.11]   [spkg-install] -- Detecting CXX compile features
    [macaulay2-1.24.11]   [spkg-install] -- Detecting CXX compile features - done
    [macaulay2-1.24.11]   [spkg-install] ## Not building from a git repository; submodules may need to be manually populated
    [macaulay2-1.24.11]   [spkg-install] ## Configure Macaulay2
    [macaulay2-1.24.11]   [spkg-install]      M2 version        = 1.24.11
    [macaulay2-1.24.11]   [spkg-install]      Git description   = release-1.24.11
    [macaulay2-1.24.11]   [spkg-install]      Git branch        = 
    [macaulay2-1.24.11]   [spkg-install]      Install prefix    = /Users/runner/sage-local
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install]      CMAKE_BUILD_TYPE  = Release
    [macaulay2-1.24.11]   [spkg-install]      BUILD_NATIVE      = ON
    [macaulay2-1.24.11]   [spkg-install]      BUILD_SHARED_LIBS = OFF
    [macaulay2-1.24.11]   [spkg-install]      BUILD_TESTING     = ON
    [macaulay2-1.24.11]   [spkg-install]      BUILD_DOCS        = OFF
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install]      COVERAGE          = OFF
    [macaulay2-1.24.11]   [spkg-install]      MEMDEBUG          = OFF
    [macaulay2-1.24.11]   [spkg-install]      PROFILING         = OFF
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install]      DEVELOPMENT       = OFF
    [macaulay2-1.24.11]   [spkg-install]      EXPERIMENT        = OFF
    [macaulay2-1.24.11]   [spkg-install] ## could not recognize issue flavor: macOS
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] ## Host OS information
    [macaulay2-1.24.11]   [spkg-install]      ISSUE             = macOS-14.7.1
    [macaulay2-1.24.11]   [spkg-install]      NODENAME          = Mac-1732423313583.local
    [macaulay2-1.24.11]   [spkg-install]      OS REL            = Darwin 23.6.0
    [macaulay2-1.24.11]   [spkg-install]      ARCH              = arm64
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] ## Staging area prefixes
    [macaulay2-1.24.11]   [spkg-install]      usr-host          = /Users/runner/sage-local/var/tmp/sage/build/macaulay2-1.24.11/src/M2/BUILD/build/usr-host
    [macaulay2-1.24.11]   [spkg-install]      common            = /Users/runner/sage-local/var/tmp/sage/build/macaulay2-1.24.11/src/M2/BUILD/build/usr-dist/common
    [macaulay2-1.24.11]   [spkg-install]      exec              = /Users/runner/sage-local/var/tmp/sage/build/macaulay2-1.24.11/src/M2/BUILD/build/usr-dist/arm64-Darwin-macOS-14.7.1
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] ## Compiler information
    [macaulay2-1.24.11]   [spkg-install]      C                 = AppleClang 15.0.0.15000309 (/Users/runner/sage-local/libexec/ccache/gcc)
    [macaulay2-1.24.11]   [spkg-install]      C++               = AppleClang 15.0.0.15000309 (/Users/runner/sage-local/libexec/ccache/g++)
    [macaulay2-1.24.11]   [spkg-install]      Ccache            = /Users/runner/sage-local/bin/ccache
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] ## CMake path variables
    [macaulay2-1.24.11]   [spkg-install]      CMAKE_PREFIX_PATH
    [macaulay2-1.24.11]   [spkg-install]        /Users/runner/sage-local/var/tmp/sage/build/macaulay2-1.24.11/src/M2/BUILD/build/usr-host
    [macaulay2-1.24.11]   [spkg-install]      CMAKE_SYSTEM_PREFIX_PATH
    [macaulay2-1.24.11]   [spkg-install]        /Users/runner/sage-local/var/tmp/sage/build/macaulay2-1.24.11/src/M2/BUILD/build/usr-host;/Applications/Xcode_15.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr;/opt/homebrew;/usr/local;/usr;/;/opt/homebrew;/Users/runner/sage-local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local
    [macaulay2-1.24.11]   [spkg-install]      CMAKE_C_IMPLICIT_LINK_DIRECTORIES
    [macaulay2-1.24.11]   [spkg-install]        /Users/runner/sage-local/lib;/Users/runner/work/passagemath/passagemath/prefix/lib;/Users/runner/sage-local/var/tmp/sage/build/macaulay2-1.24.11/src/M2;/Applications/Xcode_15.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/lib
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] --  Checking for existing libraries and programs
    [macaulay2-1.24.11]   [spkg-install] -- Found Threads: TRUE
    [macaulay2-1.24.11]   [spkg-install] CMake Warning (dev) at cmake/check-libraries.cmake:45 (find_package):
    [macaulay2-1.24.11]   [spkg-install]   Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
    [macaulay2-1.24.11]   [spkg-install]   --help-policy CMP0167" for policy details.  Use the cmake_policy command to
    [macaulay2-1.24.11]   [spkg-install]   set the policy and suppress this warning.
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] Call Stack (most recent call first):
    [macaulay2-1.24.11]   [spkg-install]   CMakeLists.txt:71 (include)
    [macaulay2-1.24.11]   [spkg-install] This warning is for project developers.  Use -Wno-dev to suppress it.
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] CMake Warning at /opt/homebrew/share/cmake/Modules/FindBoost.cmake:1414 (message):
    [macaulay2-1.24.11]   [spkg-install]   New Boost version may have incorrect or missing dependencies and imported
    [macaulay2-1.24.11]   [spkg-install]   targets
    [macaulay2-1.24.11]   [spkg-install] Call Stack (most recent call first):
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindBoost.cmake:1539 (_Boost_COMPONENT_DEPENDENCIES)
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindBoost.cmake:2151 (_Boost_MISSING_DEPENDENCIES)
    [macaulay2-1.24.11]   [spkg-install]   cmake/check-libraries.cmake:45 (find_package)
    [macaulay2-1.24.11]   [spkg-install]   CMakeLists.txt:71 (include)
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] CMake Warning at /opt/homebrew/share/cmake/Modules/FindBoost.cmake:1414 (message):
    [macaulay2-1.24.11]   [spkg-install]   New Boost version may have incorrect or missing dependencies and imported
    [macaulay2-1.24.11]   [spkg-install]   targets
    [macaulay2-1.24.11]   [spkg-install] Call Stack (most recent call first):
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindBoost.cmake:1539 (_Boost_COMPONENT_DEPENDENCIES)
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindBoost.cmake:2151 (_Boost_MISSING_DEPENDENCIES)
    [macaulay2-1.24.11]   [spkg-install]   cmake/check-libraries.cmake:45 (find_package)
    [macaulay2-1.24.11]   [spkg-install]   CMakeLists.txt:71 (include)
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] CMake Warning at /opt/homebrew/share/cmake/Modules/FindBoost.cmake:1414 (message):
    [macaulay2-1.24.11]   [spkg-install]   New Boost version may have incorrect or missing dependencies and imported
    [macaulay2-1.24.11]   [spkg-install]   targets
    [macaulay2-1.24.11]   [spkg-install] Call Stack (most recent call first):
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindBoost.cmake:1539 (_Boost_COMPONENT_DEPENDENCIES)
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindBoost.cmake:2151 (_Boost_MISSING_DEPENDENCIES)
    [macaulay2-1.24.11]   [spkg-install]   cmake/check-libraries.cmake:45 (find_package)
    [macaulay2-1.24.11]   [spkg-install]   CMakeLists.txt:71 (include)
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] CMake Error at /opt/homebrew/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
    [macaulay2-1.24.11]   [spkg-install]   Could NOT find Boost (missing: regex) (found version "1.86.0")
    [macaulay2-1.24.11]   [spkg-install] Call Stack (most recent call first):
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
    [macaulay2-1.24.11]   [spkg-install]   /opt/homebrew/share/cmake/Modules/FindBoost.cmake:2409 (find_package_handle_standard_args)
    [macaulay2-1.24.11]   [spkg-install]   cmake/check-libraries.cmake:45 (find_package)
    [macaulay2-1.24.11]   [spkg-install]   CMakeLists.txt:71 (include)
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] 
    [macaulay2-1.24.11]   [spkg-install] -- Configuring incomplete, errors occurred!
    [macaulay2-1.24.11]   [spkg-install] ********************************************************************************
    [macaulay2-1.24.11]   [spkg-install] Error configuring macaulay2-1.24.11 with cmake
```
https://github.com/passagemath/passagemath/actions/runs/11993290637/job/33434249651?pr=461#step:11:2117

--- a/build/pkgs/boost_cropped/spkg-install.in
+++ b/build/pkgs/boost_cropped/spkg-install.in
@@ -1,2 +1,3 @@
-echo "Copying over new boost headers"
-sdh_install src/boost "$SAGE_LOCAL/include"
+cd src
+./bootstrap.sh --prefix="$SAGE_LOCAL" --with-libraries=regex
+./b2 install

